src/vellum/workflows/events/types.py

```diff
@@ -68,20 +69,30 @@ class PromptDeploymentParentContext(BaseDeploymentParentContext):
 
 class NodeParentContext(BaseParentContext):
     type: Literal["WORKFLOW_NODE"] = "WORKFLOW_NODE"
-    node_definition: Type["BaseNode"]
+    node_definition: Union[Type["BaseNode"], CodeResourceDefinition]
```

Comment:
Oh boy. I _think_ I know why we need to do this but... would love to sync on this

Reply:
actually I don't know if we could avoid this...

Reply:
discussed offline - yea we need `CodeResourceDefinition`. To make our lives easier, let's just commit to the latter